impeller/renderer/backend/vulkan/pipeline_cache_vk.cc

```diff
@@ -105,7 +106,11 @@ PipelineCacheVK::PipelineCacheVK(std::shared_ptr<const Capabilities> caps,
   is_valid_ = !!cache_;
 }
 
-PipelineCacheVK::~PipelineCacheVK() = default;
+PipelineCacheVK::~PipelineCacheVK() {
+  if (!device_holder_.lock()) {
+    cache_.release();
```

Comment:
If `device_holder_.lock()` succeeds, then call `cache_.reset()` within that block.

This avoids a potential scenario where the Vulkan device is destroyed after `device_holder_.lock()` goes out of scope but before the members of `PipelineCacheVK` are destructed.

(This also applies to the `PipelineVK` and `ShaderFunctionVK` destructors)

Reply:
Good call.  We have to do more than that though.  We have to hold onto the lock for the whole duration of the branch.  I made both changes.